iothub/device/src/Transport/Mqtt/MqttTransportHandler.cs

```diff
@@ -419,14 +419,9 @@ public override async Task SendTelemetryAsync(TelemetryMessage message, Cancella
             }
         }
 
-        public override async Task SendTelemetryBatchAsync(IEnumerable<TelemetryMessage> messages, CancellationToken cancellationToken)
+        public override Task SendTelemetryBatchAsync(IEnumerable<TelemetryMessage> messages, CancellationToken cancellationToken)
         {
-            cancellationToken.ThrowIfCancellationRequested();
-
-            // Note that this sends all messages at once and then waits for all the acknowledgements. This
-            // is the recommended pattern for sending large numbers of messages over an asynchronous
-            // protocol like MQTT
-            await Task.WhenAll(messages.Select(x => SendTelemetryAsync(x, cancellationToken))).ConfigureAwait(false);
+            throw new InvalidOperationException("Batch telemetry is supported only over AMQP.");
```

Comment:
I feel like the MQTT transport layer shouldn't know about the other transports. A more cohesive message would be that it is not supported over MQTT. Thoughts?

Reply:
I see your point. I think we'll still need to guide users down the correct path.
How about:
`throw new InvalidOperationException("This operation is not supported over this protocol. Please refer to the doc comments for additional details.");`